content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+Putting your learnr tutorial in a package is also free-- and always will be, which means you can also sidestep costs associated with hosting services! Your package doesn't have to be on CRAN, and it doesn't have to contain functions or data (but it can if you want!).
+
+Never made a package before? No sweat! In this post, we'll guide you through the process.
+
+
+
+## tl;dr
+
+Here's how you put an interactive learnr tutorial in a package:
+
+1. In a GitHub-connected RStudio project, run `usethis::create_package(getwd())` to make a basic package.
+
+1. Run `usethis::use_tutorial("<name-of-learnr-file>", "<Title You'd Like the User to See>")` to create the learnr tutorial. Customize with your content. Repeat for additional tutorials.
+
+1. Go to *Build* > *Install and Restart*.
+
+1. Optionally, run `usethis::use_package("gradethis")` to add packages dependencies for exercise-checking.
+
+1. Optionally, edit the `DESCRIPTION` file.
+
+1. Run `usethis::use_readme_rmd()` to add a README. Knit when you're done.
+
+1. Push to GitHub and have folks install with `remotes::install_github("<your-repo>/<package-name>")`
+
+1. Repeat steps 2, 3, and 7 anytime you update your tutorial package.  
+
+
+```{r, out.width = "100%", echo = FALSE}
+knitr::include_graphics("pkg-illo.jpg")
+```
+
+## How to make a package for a learnr tutorial
+
+
+We will walk through the steps for creating the bare-minimum package that you can use to house and deliver your interactive learnr tutorials to your users.
+
+Our final goal is to create a package that is "downloadable" by users, which means this package needs a GitHub repository so that it can be easily shared. If students wanted to, they’d be able to find the source code and solutions in your GitHub repository (or, with some spelunking, in the folder that houses all their R packages locally). But this would take some extra effort on the part of students.
+
+If a solution like this still works for your purposes, then read onwards!
+
+
+### Pre-requisites
+
+We assume the following:
+
+* You have a GitHub account (but you don't need to be more than a GitHub novice)
+* You are comfortable with the basics of using R and RStudio
+* You understand how R projects work
+* You are already familiar with creating a regular learnr tutorial
+
+
+
+### Getting set up
+
+Let's make sure we have all the pieces in place to create a learnr tutorial package.
+
+
+#### Packages
+
+Make sure you've installed and loaded the following packages:
+
+```{r, eval = FALSE}
+#install.packages(c("usethis", "devtools", "roxygen2"))
+library(usethis)
+library(roxygen2)
+library(devtools)
+library(learnr)
+```
+
+
+#### Create a GitHub repository
```

Comment:
This is a follow on from my comment above re `use_git()` and `use_github()` -- since we're saying they should use usethis to create the package and usethis offers these functions, I think the repo first approach makes more sense here. Otherwise, it's a bit clunky to clone a repo, get into it, and then create the package there.